src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/InitializationService.java

```diff
@@ -0,0 +1,315 @@
+
+    @Setter
+    private boolean isTokenPinEnforced = SystemProperties.shouldEnforceTokenPinPolicy();
+
+    @Autowired
+    public InitializationService(SystemService systemService, ServerConfService serverConfService,
+            TokenService tokenService, GlobalConfFacade globalConfFacade, ClientService clientService,
+            SignerProxyFacade signerProxyFacade) {
+        this.systemService = systemService;
+        this.serverConfService = serverConfService;
+        this.tokenService = tokenService;
+        this.globalConfFacade = globalConfFacade;
+        this.clientService = clientService;
+        this.signerProxyFacade = signerProxyFacade;
+    }
+
+    /**
+     * Check the whole init status of the Security Server. The init status consists of the following:
+     * 1. is anchor imported - whether or not a configuration anchor has been imported
+     * 2. is server conf initialized - whether or not a server conf exists
+     * 3. is software token initialized - whether or not a software token exists AND
+     * it's status != TokenStatusInfo.NOT_INITIALIZED
+     * @return
+     */
+    public InitializationStatusDto isSecurityServerInitialized() {
+        boolean isAnchorImported = systemService.isAnchorImported();
+        boolean isServerConfInitialized = serverConfService.isServerConfInitialized();
+        boolean isSoftwareTokenInitialized = tokenService.isSoftwareTokenInitialized();
+        InitializationStatusDto initializationStatusDto = new InitializationStatusDto();
+        initializationStatusDto.setAnchorImported(isAnchorImported);
+        initializationStatusDto.setServerConfInitialized(isServerConfInitialized && isSoftwareTokenInitialized);
+        return initializationStatusDto;
+    }
+
+    /**
+     * Initialize a new Security Server with the provided parameters. The method will throw an exception if the
+     * server has already been initialized. The method does support partial initialization (e.g. server conf has
+     * been created but software token has not) BUT taking that into use will require changing
+     * {@link #verifyInitializationPrerequisites()} to throw UnhandledWarningsException with warning metadata
+     * instead of InitializationException.
+     * @param securityServerCode server code for the new Security Server
+     * @param ownerMemberClass member class of the new owner member
+     * @param ownerMemberCode member code of the new owner member
+     * @param softwareTokenPin pin code for the initial software token (softToken-0)
+     * @param ignoreWarnings
+     * @throws AnchorNotFoundException if an anchor has not been imported
+     * @throws InitializationException if prerequisite check fails: if a server conf already exists OR if a
+     * software token has already been initialized
+     * @throws UnhandledWarningsException
+     * @throws WeakPinException if the pin does not meet the length and complexity requirements (if token pin policy is
```

Comment:
Had some trouble understanding `WeakPinException` and `InvalidPinException` initially. Reading the exception names and javadocs made me think "Isn't a weak pin also an invalid PIN? And what is InvalidPinException about, if not length and complexity requirements".

I guess the idea is that frontend will be able to show proper error messages based on classification to error types `invalid_pin` and `weak_pin`? And that there will likely be two different error messages? 

The difference between invalid_pin and weak_pin does not seem too obvious. Could also do some renaming and updating of javadocs to make it more clear? To me it looks like _bad characters_ in PIN is invalid_pin and _too few characters or too few character classes_ is weak_pin. But it took a few rereads to understand that. 

There's also the possibility of using one single exception class InvalidPinException and using the same metadata-based way of specifying how exactly was it invalid as is already used for weak / METADATA_PIN_MIN_LENGTH and weak / METADATA_PIN_MIN_CHAR_CLASSES. I am not sure if it would be better or worse.

Reply:
I just renamed the `InvalidPinException` to `InvalidCharactersException` and added javadocs for further explanation